gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeTableOperationsTest.java

```diff
@@ -0,0 +1,200 @@
+
+  @BeforeEach
+  public void before() throws Exception {
+    ImmutableMap<String, String> properties =
+        ImmutableMap.of(
+            GCPProperties.PROJECT_ID,
+            GCP_PROJECT,
+            GCPProperties.REGION,
+            GCP_REGION,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            temp.toFile().getAbsolutePath(),
+            GCPProperties.BIGLAKE_CATALOG_ID,
+            CATALOG_ID);
+
+    bigLakeCatalog = new BigLakeCatalog();
+    bigLakeCatalog.setConf(new Configuration());
+    bigLakeCatalog.initialize(CATALOG_NAME, properties, bigLakeClient);
+    tableOps = (BigLakeTableOperations) bigLakeCatalog.newTableOps(TABLE_IDENTIFIER);
+  }
+
+  @AfterEach
+  public void after() throws Exception {
+    bigLakeCatalog.close();
+  }
+
+  @Test
+  public void testDoCommitShouldUseEtagForUpdateTable() throws Exception {
+    when(bigLakeClient.getTable(TABLE_NAME))
+        .thenThrow(new NoSuchTableException("error message getTable"));
+    Table createdTable = createTestTable();
+
+    Table tableWithEtag = createdTable.toBuilder().setEtag("etag").build();
+    reset(bigLakeClient);
+    when(bigLakeClient.getTable(TABLE_NAME)).thenReturn(tableWithEtag, tableWithEtag);
+
+    org.apache.iceberg.Table loadedTable = bigLakeCatalog.loadTable(TABLE_IDENTIFIER);
+
+    when(bigLakeClient.updateTableParameters(any(), any(), any())).thenReturn(tableWithEtag);
+    loadedTable.updateSchema().addColumn("n", Types.IntegerType.get()).commit();
+
+    ArgumentCaptor<TableName> nameCaptor = ArgumentCaptor.forClass(TableName.class);
+    ArgumentCaptor<String> etagCaptor = ArgumentCaptor.forClass(String.class);
+    verify(bigLakeClient, times(1))
+        .updateTableParameters(nameCaptor.capture(), any(), etagCaptor.capture());
+    assertThat(nameCaptor.getValue()).isEqualTo(TABLE_NAME);
+    assertThat(etagCaptor.getValue()).isEqualTo("etag");
+  }
+
+  @Test
+  public void testDoCommitShouldFailWhenEtagMismatch() throws Exception {
+    when(bigLakeClient.getTable(TABLE_NAME))
+        .thenThrow(new NoSuchTableException("error message getTable"));
+    Table createdTable = createTestTable();
+
+    Table tableWithEtag = createdTable.toBuilder().setEtag("etag").build();
+    reset(bigLakeClient);
+    when(bigLakeClient.getTable(TABLE_NAME)).thenReturn(tableWithEtag, tableWithEtag);
+
+    org.apache.iceberg.Table loadedTable = bigLakeCatalog.loadTable(TABLE_IDENTIFIER);
+
+    when(bigLakeClient.updateTableParameters(any(), any(), any()))
+        .thenThrow(
+            new AbortedException(
+                new RuntimeException("error message etag mismatch"),
+                GrpcStatusCode.of(Code.ABORTED),
+                false));
+
+    assertThatThrownBy(
+            () -> loadedTable.updateSchema().addColumn("n", Types.IntegerType.get()).commit())
+        .isInstanceOf(CommitFailedException.class);
+  }
+
+  @Test
+  public void testDoFreshRefreshShouldReturnNullForNonIcebergTable() {
```

Comment:
is this called on purpose a "fresh refresh"? Maybe something like "initial refresh"?

Reply:
Done.